lib/util/Components.js

```diff
@@ -70,7 +94,16 @@ Components.prototype.set = function(node, props) {
     return;
   }
   const id = this._getId(node);
+  let copyUsedPropTypes;
+  if (this._list[id]) {
+    // usedPropTypes is an array. _extend replaces existing array with a new one which caused issue #1309.
```

Comment:
@ljharb added a comment here with some explanation for the change. LMK if there is a better place for it.